### PR TITLE
feat(eks/cijenkinsio-agents-2): setup technical addons for taint and toleration + add autoscaler

### PR DESCRIPTION
**Pull request by smerle33**
Ref: https://github.com/jenkins-infra/helpdesk/issues/4319

adding the addons with taints and toleration
adding autoscaler with helm provider

### CODE REVIEW

**Comment by dduportal** [comment]
Added a few changes:

- Removed the `kube-proxy` add-on configuration to avoid the error `InvalidParameterException: ConfigurationValue provided in request is not supported: Json schema validation failed with error: [$.tolerations: is not defined in the schema and the schema does not allow additional properties]`
- Factorized the tolerations/taints and node labels to locals to avoid repetition and human error
- Fixed the `helm_release` resource to:
  - Use the alias provider (otherwise it cannot find which kubernetes config to use as there are no default helm provider)
  - Use the namespace specified for the service account (otherwise the installation won't be able to authenticate properly from the `default` namespace). Need to create the namespace if not existing (setting up the associated attribute to true)
  - Pass tolerations and node selector labels to the templatized helm values
  - Make the templated YAML safer by using double quote on variable (to avoid accidental implicit castings to booleans or numbers)

File: helm/cluster-autoscaler-values.yaml.tfpl [hunk 1]
```diff
@@ -0,0 +1,25 @@
+---
+awsRegion: ${region}
```
Comment (smerle33) [inline helm/cluster-autoscaler-values.yaml.tfpl:2]:
```suggestion
awsRegion: "${region}"
```

File: helm/cluster-autoscaler-values.yaml.tfpl [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+nodeSelector:
+%{ for label_key, label_value in nodeSelectors ~}
+  ${label_key}: "${label_value}"
+%{ endfor ~}
```
Comment (smerle33) [inline helm/cluster-autoscaler-values.yaml.tfpl:7]:
```suggestion
nodeSelector:
${yamlencode(nodeSelectors)}
```

File: locals.tf [hunk 3]
```diff
@@ -1,17 +1,39 @@
...
-  ci_jenkins_io_fqdn = "aws.ci.jenkins.io"
+  ci_jenkins_io = {
+    service_fqdn       = "ci.jenkins.io"
+    controller_vm_fqdn = "aws.ci.jenkins.io"
+  }
+
+  cijenkinsio_agents_2 = {
+    autoscaler = {
+      namespace      = "autoscaler",
+      serviceaccount = "cluster-autoscaler-aws-cluster-autoscaler-chart",
```
Comment (smerle33) [inline locals.tf:18]:
```suggestion
      serviceaccount = "autoscaler",
```

**Review by dduportal — APPROVED** [review]
`Plan: 5 to add, 1 to change, 1 to destroy.`

**Pull request merged**